internal/rpc/server/ospkg/server_test.go

```diff
@@ -0,0 +1,146 @@
+package ospkg
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	ospkg2 "github.com/aquasecurity/trivy/pkg/detector/ospkg"
+
+	"github.com/aquasecurity/trivy/pkg/log"
+
+	"golang.org/x/xerrors"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/stretchr/testify/mock"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/aquasecurity/trivy/pkg/types"
+	"github.com/aquasecurity/trivy/pkg/vulnerability"
+	proto "github.com/aquasecurity/trivy/rpc/detector"
+)
+
+func TestMain(m *testing.M) {
+	log.InitLogger(false, false)
+	code := m.Run()
+	os.Exit(code)
+}
+
+func TestServer_Detect(t *testing.T) {
+	type detectInput struct {
```

Comment:
Here and in previous test you have a very similar strucs to hold input and return args passed to mocks. You can consider having a generic structure to describe an expected behaviour of the given mock, see https://github.com/aquasecurity/harbor-scanner-trivy/blob/master/pkg/mock/expectation.go

```
// Expectation represents an expectation of a method being called and its return values.
type Expectation struct {
	Method     string
	Args       []interface{}
	ReturnArgs []interface{}
}

// ApplyExpectation applies the specified expectations on a given mock.
func ApplyExpectations(t *testing.T, mock interface{}, expectations ...*Expectation) {
}
```

This might improve readability of the test cases.

Reply:
Your approach looks so cool, but I'd like to keep named args. I refactored here. Let me know your thought, please. If it looks fine, I will try to refactor other places.
https://github.com/aquasecurity/trivy/pull/303/commits/99ef6ecd434c1fd2fcae8af377ff92490c096b65

Reply:
I like having a struct for mock expectations as you did in `detector_mock.go`